lms/djangoapps/courseware/model_data.py

```diff
@@ -34,12 +35,473 @@ class InvalidWriteError(Exception):
+        else:
+            return field_object.modified
+
+    @contract(kv_dict="dict(DjangoKeyValueStore_Key: *)")
+    def set_many(self, kv_dict):
+        """
+        Set the specified fields to the supplied values.
+
+        Arguments:
+            kv_dict (dict): A dictionary mapping :class:`~DjangoKeyValueStore.Key`
+                objects to values to set.
+        """
+        pending_updates = defaultdict(dict)
+        for kvs_key, value in kv_dict.items():
+            cache_key = self._cache_key_for_kvs_key(kvs_key)
+            field_state = self._cache[cache_key]
+            field_state[kvs_key.field_name] = value
+
+            pending_updates[cache_key][kvs_key.field_name] = value
+
+        try:
+            self._client.set_many(
+                self.user.username,
+                pending_updates
+            )
+        except DatabaseError:
+            raise KeyValueMultiSaveError(sum((state.keys() for state in pending_updates.values()), []))
+
+    @contract(kvs_key=DjangoKeyValueStore.Key)
+    def get(self, kvs_key):
+        """
+        Return the django model object specified by `kvs_key` from
+        the cache.
+
+        Arguments:
+            kvs_key (`DjangoKeyValueStore.Key`): The field value to delete
+
+        Returns: A django orm object from the cache
+        """
+        cache_key = self._cache_key_for_kvs_key(kvs_key)
+        if cache_key not in self._cache:
+            raise KeyError(kvs_key.field_name)
+
+        return self._cache[cache_key][kvs_key.field_name]
+
+    @contract(kvs_key=DjangoKeyValueStore.Key)
+    def delete(self, kvs_key):
+        """
+        Delete the value specified by `kvs_key`.
+
+        Arguments:
+            kvs_key (`DjangoKeyValueStore.Key`): The field value to delete
+
+        Raises: KeyError if key isn't found in the cache
+        """
+        cache_key = self._cache_key_for_kvs_key(kvs_key)
+        if cache_key not in self._cache:
+            raise KeyError(kvs_key.field_name)
+
+        field_state = self._cache[cache_key]
+
+        # If field_name isn't in field_state,
+        # then this will throw the correct KeyError
+        del field_state[kvs_key.field_name]
+
+        self._client.delete(self.user.username, cache_key, fields=[kvs_key.field_name])
+
+    @contract(kvs_key=DjangoKeyValueStore.Key, returns=bool)
+    def has(self, kvs_key):
+        """
+        Return whether the specified `kvs_key` is set.
+
+        Arguments:
+            kvs_key (`DjangoKeyValueStore.Key`): The field value to delete
+
+        Returns: bool
+        """
+        cache_key = self._cache_key_for_kvs_key(kvs_key)
+
+        return (
+            cache_key in self._cache and
+            kvs_key.field_name in self._cache[cache_key]
+        )
+
+    @contract(user_id=int, usage_key=UsageKey, score="number|None", max_score="number|None")
+    def set_score(self, user_id, usage_key, score, max_score):
+        """
+        UNSUPPORTED METHOD
```

Comment:
It's unsupported because we want to move score-recording behind a different, separate interface.